src/mono/mono.proj

```diff
@@ -466,10 +480,10 @@
         <Destination>$(RuntimeBinDir)\opt</Destination>
       </_MonoRuntimeArtifacts>
       <_MonoRuntimeArtifacts Condition="'$(MonoAOTBundleLLVMOptimizer)' == 'true'" Include="$(MonoAOTLLVMDir)\bin\llc">
-        <Destination>$(RuntimeBinDir)cross\llc</Destination>
+        <Destination>$(RuntimeBinDir)cross\$(TargetOS)-$(TargetArchitecture)\llc</Destination>
       </_MonoRuntimeArtifacts>
```

Comment:
llc/opt are not target arch specific.

Reply:
Assuming that this isn't true is easier for now, i.e. put "the `llc/opt` needed for this `mono-aot-cross` in the same bundle" rather than dependency tracking & public artifact publishing of the llvm binaries for end user consumption. We can clean that story up later.